### DISCUSSION TITLE
How to edit my applications commands permission?

### DISCUSSION THREAD

**Discussion by yuhime** [Rapptz/discord.py#9230] [Q&A]
I need to set commands in my application so that they are visible only to administrators and only for a specific role and channel, but I can't figure out how to do it. In a nutshell, I need to customize the integration of the commands and I would like to do it directly from the bot. I have seen another bot do it (MEE6), so I think it is possible.

**Comment by AbstractUmbra** [comment]
The only provided way to do this is with the [`default_permissions`](https://discordpy.readthedocs.io/en/stable/interactions/api.html?highlight=default_permissions#discord.app_commands.default_permissions) decorator.
You cannot "hide" (make non-visible in the UI) from channels or roles, only from guild level permissions given via roles.

The documentation link above has examples and a more in-depth explanations.
